src/shared/bpf-dlopen.c

```diff
@@ -181,6 +181,19 @@ int dlopen_bpf(void) {
         return r;
 }
 
+int bpf_get_error_translated(const void *ptr) {
+        int r;
+
+        r = sym_libbpf_get_error(ptr);
+
+        switch (r) {
+        case -524:
```

Comment:
Sorry, but where does this 524 come from? Isn't this something that should be fixed in libbpf?

Reply:
From the kernel, and it looks like it will not be fixed, at least not without some new API (that we'll need to shim for backward compat): https://github.com/systemd/systemd/issues/32170#issuecomment-2135769555

Reply:
This really should be fixed in the kernel I am sure, they should not let this into userspace.

Can you add a comment about the 524, with a link to the kernel's internal header file, i.e.:

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/errno.h?h=v6.9&id=a38297e3fb012ddfa7ce0321a7e5a8daeb1872b6#n27

And in particular, can you add a comment that this is a bug in the kernel we are working around here that we really would prefer if the kernel would fix itself.